kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/HasMetadata.java

```diff
@@ -232,4 +235,109 @@ default boolean isFinalizerValid(String finalizer) {
   default boolean removeFinalizer(String finalizer) {
     return getMetadata().getFinalizers().remove(finalizer);
   }
+
+  default boolean hasOwnerReferenceFor(HasMetadata owner) {
+    return getOwnerReferenceFor(owner).isPresent();
+  }
+
+  default boolean hasOwnerReferenceFor(String ownerUid) {
+    return getOwnerReferenceFor(ownerUid).isPresent();
+  }
+
+  default Optional<OwnerReference> getOwnerReferenceFor(HasMetadata owner) {
+    if (owner == null) {
+      return Optional.empty();
+    }
+
+    final String ownerUID = owner.getMetadata().getUid();
```

Comment:
Calls to getMetadata() on passed in objects and even this object can be null, is that assumed to be a usage error?

Reply:
How often does that occur, though? We could shield against this case but we already have code that assumes the metadata to exist and it hasn't caused issues so far (that I'm aware of, at least). If we decide to shield against `null` metadata, we should fix the other usages as well.

Reply:
The rest of the fabric8 code is pretty mixed with checking for null metadata in many places it's not needed as the object is coming from the api-server.  But anytime it can be passed in there's a chance it could be null, which is typically guarded against in BaseOperation methods.  It's of course optional to add checks here, it's just a question of whether it's worth the effort to have something other than a NPE.